src/executeCommandLine/executeCommandLine.ts

```diff
@@ -90,32 +90,53 @@ namespace ts {
     }
 
     function createColors(sys: System) {
-        const showColors = defaultIsPretty(sys);
+        const showColors = defaultIsPretty(sys) || !!sys.getEnvironmentVariable("NO_COLOR");

```

Comment:
This looks wrong. Shouldn't it be `defaultIsPretty(sys) && !sys.getEnvironmentVariable("NO_COLOR")`? If `NO_COLOR` exists, then we shouldn't print color... as is, if `NO_COLOR` is set, then tsc will write with colors even if it is not printing to a terminal.

(There's also the consideration that we should be checking for existence, not non-emptyness, but that's less important)

Reply:
Coming back to this - I think the environment check belongs in the `defaultIsPretty` function... but maybe I don't understand that yet.

Reply:
Thanks, yeah I was avoiding making more drastic changes by adding NO_COLOR support everywhere (effectively making this bug-fix a feature) by moving it into `defaultIsPretty` - running the tests now too see how that changes the test suite.